### PR TITLE
Fully annotate `pyvista_ndarray.py`

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Add annotations.

### CODE REVIEW

File: pyvista/core/pyvista_ndarray.py [hunk 1]
```diff
@@ -111,7 +118,7 @@ def __setitem__(self, key: int | NumpyArray[int], value) -> None:
         if dataset is not None and dataset.Get():
             dataset.Get().Modified()
 
-    def __array_wrap__(self, out_arr, context=None, return_scalar: bool = False):
+    def __array_wrap__(self: pyvista_ndarray, out_arr, context=None, return_scalar: bool = False):  # noqa: ANN001, ANN204
```
Comment (tkoyama010) [inline pyvista/core/pyvista_ndarray.py:121]: Could you explain us why we don't define the type of `out_arr` ?

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/core/pyvista_ndarray.py (hunk 1 above)
Comment (user27182) [reply pyvista/core/pyvista_ndarray.py:121]: My IDE (PyCharm) gives this type hint by default:
array_wrap

It looked a bit too complex to reproduce, and I'm not sure these annotations would be much benefit in terms of external API type hints for users.

Comment (tkoyama010) [reply pyvista/core/pyvista_ndarray.py:121]: Got it. Thanks!

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**